### ISSUE TITLE
Can we refrain from creating global variables? 

### ISSUE THREAD

**Issue by pcgeek86** [Azure/azure-powershell#2088]
https://github.com/Azure/azure-powershell/blob/8bcf538af27c06588bf75854a99c309a43b21744/src/ResourceManager/Profile/CheckVersions.ps1

```
PS C:\Users\TrevorSullivan> Get-Variable -Name AzureRMModules, AvailableModules | Format-List


PSPath        : Microsoft.PowerShell.Core\Variable::AzureRMModules
PSDrive       : Variable
PSProvider    : Microsoft.PowerShell.Core\Variable
PSIsContainer : False
Name          : AzureRMModules
Description   : 
Value         : {AzureRM.ApiManagement, AzureRM.Automation, AzureRM.Backup, AzureRM.Batch...}
Visibility    : Public
Module        : 
ModuleName    : 
Options       : None
Attributes    : {}

PSPath        : Microsoft.PowerShell.Core\Variable::AvailableModules
PSDrive       : Variable
PSProvider    : Microsoft.PowerShell.Core\Variable
PSIsContainer : False
Name          : AvailableModules
Description   : 
Value         : {AWSPowerShell, AWSPowerShell, AWSPowerShell, Azure.Storage...}
Visibility    : Public
Module        : 
ModuleName    : 
Options       : None
Attributes    : {}
```

**Comment by pcgeek86** [comment]
@markcowl @ravbhatnagar Please advise.

**Comment by markcowl** [comment]
@pcgeek86 Will do.

**Comment by markcowl** [comment]
@pcgeek86 Fixed by: #2183
